src/python/database.py

```
@@ -545,28 +520,24 @@ def link_task_label(self, task_id, label_id):
-                task = self.get_task(task_id)
-                if task is None:
-                    raise DatabaseError(f"No task found with ID {task_id}", "TASK_NOT_FOUND")
-
-                # Check if label exists
-                label = self.get_label(label_id)
-                if label is None:
-                    raise DatabaseError(f"No label found with ID {label_id}", "LABEL_NOT_FOUND")
+                try:
+                    cursor.execute(query, (task_id, label_id))
+                except sqlite3.IntegrityError:
+                    raise DatabaseError("Task-label link already exists", "LINK_EXISTS")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)